### PR TITLE
Use torch fx graph DCE

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #138950
* __->__ #138932
* #138868

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

### CODE REVIEW

File: torch/fx/passes/_tensorify_python_scalars.py [hunk 1]
```diff
@@ -261,10 +248,9 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
 
                     node.replace_all_uses_with(replacement_proxy.node)
                     graph.erase_node(node)
-                    deleted.add(node)
 
     # DCE symbols (which are guaranteed to be pure) only
-    dce(graph, deleted)
+    graph.eliminate_dead_code()
```
Comment (ezyang) [inline torch/fx/passes/_tensorify_python_scalars.py:253]: @bdhirsh do you know if it's safe to call `eliminate_dead_code()` on arbitrary FX graphs now. I still feel uncomfortable about the default predicate

Comment (ezyang) [reply torch/fx/passes/_tensorify_python_scalars.py:253]: (Graph cannot be assumed to be functionalized)

Comment (bobrenjc93) [reply torch/fx/passes/_tensorify_python_scalars.py:253]: Why can't graphs can't be assumed to functionalized here? This pass is after functionalization and in our pass we only rewrite from TensorScalar to ScalarScalar and node.target(args) to node.target(args_with_symfloats_specialized)? Both of which won't introduce mutations?

Comment (ezyang) [reply torch/fx/passes/_tensorify_python_scalars.py:253]: Oh you might be right lol.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**